optuna_dashboard/ts/components/GraphContour.tsx

```diff
@@ -17,15 +17,25 @@ import { useMergedUnionSearchSpace } from "../searchSpace"
 import { getAxisInfo } from "../graphUtil"
 import { useQuery } from "../urlQuery"
 import { getPlotAPI, PlotType } from "../apiClient"
+import { useRecoilValue } from "recoil"
+import { plotlypyIsAvailableState } from "../state"
 
 const plotDomId = "graph-contour"
 
 export const Contour: FC<{
   study: StudyDetail | null
 }> = ({ study = null }) => {
   const query = useQuery()
+  const plotlypyIsAvailable = useRecoilValue<boolean>(plotlypyIsAvailableState)
   if (query.get("plotlypy_rendering") === "true") {
-    return <ContourBackend study={study} />
+    if (plotlypyIsAvailable) {
+      return <ContourBackend study={study} />
+    } else {
+      console.warn(
+        "Use frontend rendering because plotlypy is specified but not available."
+      )
+      return <ContourFrontend study={study} />
+    }
```

Comment:
Perhaps, we could define the following wrapper function in `state.ts`:

```js
export const usePlotlypyIsAvailable = (): boolean => {
  const query = useQuery()
  const plotlypyIsAvailable = useRecoilValue<boolean>(plotlypyIsAvailableState)

  if (query.get("plotlypy_rendering") === "true") {
    if (plotlyPyIsAvailable) {
      return true
    }
    console.warn(...)
  }
  return false
}
```

Reply:
Furthermore, we could use notistack to show a warning message to users like below:
https://github.com/optuna/optuna-dashboard/blob/8033e79da6de5a4a9036a9375083ed93cc291772/optuna_dashboard/ts/action.ts#L234-L236

Reply:
Yeah, I will introduce this function when I implement backend rendering of  other graphs 👍